processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/MapParserImpl.java

```diff
@@ -73,9 +73,12 @@ public ArrayObject parse(Object data) {
 
   @Override
   public ArrayObject parseRaw(Object data) {
-    Object keyArray = ((Object[]) data)[0];
-    Object valueArray = ((Object[]) data)[1];
-    return new ArrayObject(new Object[]{child.parseRaw(keyArray), child.parseRaw(valueArray)});
+    Object[] keyValuePairs = ((Object[]) data);
```

Comment:
What is the reason for this change?

Reply:
Here data is a list of key-value pairs. So both the variables keyArray and valueArray contained (key,value) pairs in each which was wrong. So made the required changes.

Reply:
okay